src/scheduler/pct.rs

```diff
@@ -100,6 +97,13 @@ impl Scheduler for PctScheduler {
         // > enables/creates a second thread.
         // TODO is this really correct? need to think about it more
         if runnable.len() > 1 {
+            // If any new tasks were created, assign them priorities at random
+            let known_tasks = self.priority_queue.len();
+            let max_tasks = usize::from(*runnable.iter().max().unwrap());
+            for tid in known_tasks..max_tasks {
+                let index = self.rng.gen_range(1, self.priority_queue.len() + 1);
+                self.priority_queue.insert(index, TaskId::from(tid));
+            }
```

Comment:
I think maybe we need to do this outside the `if runnable.len() > 1` check? It's possible that `runnable` contains a single task ID that we happen not to have seen before. I can't think of a way our current synchronization primitives would allow this (task spawns don't immediately block the spawner), but we should do the right thing anyway.

Reply:
Good call, will fix.